File: Ejercicio3.hs
module Ejercicio3 where

import System.IO


dividir :: Double -> Double -> Either String Double
dividir num den
    | den == 0  = Left "Error: División por cero"
    | otherwise = Right (num / den)

main :: IO ()
main = do
    hSetBuffering stdout NoBuffering
    putStr "Ingrese el numerador (número): "
    numStr <- getLine
    putStr "Ingrese el denominador (número): "
    denStr <- getLine
    let num = read numStr :: Double
    let den = read denStr :: Double
    let resultado = dividir num den
    case resultado of
        Left err -> putStrLn err
        Right val -> putStrLn $ "Resultado de la división: " ++ show val

File: Ejercicio3.py


def calculadora_division():
    try:
        num1 = float(input("Ingresa el primer número (numerador): "))
        num2 = float(input("Ingresa el segundo número (denominador): "))

        if num2 == 0:
            print("Error: No se puede dividir por cero.")
        else:
            resultado = num1 / num2
            print(f"El resultado de la división es: {resultado}")
    except ValueError:
        print("Entrada inválida. Por favor, ingresa números válidos.")

if __name__ == "__main__":
    calculadora_division()


